core/modules/replica/ReplicaInfo.cc

```diff
@@ -173,6 +193,24 @@ std::map<std::string,ReplicaInfo::FileInfo> ReplicaInfo::fileInfoMap() const {
     return result;
 }
 
+bool ReplicaInfo::equalFileCollections(ReplicaInfo const& other) const {
+
+    // Files of both comllection needs to be map-sorted because objects may
```

Comment:
collection

You could also use `std::set` for ordering, but you need `operator<` for FileInfo